Move Block Explorer to footer

carletex opened the pull request:
For the sake of cleaning up the UI a bit (Example UI might be next :D Will create an issue soon), I'm testing moving the Block Explorer from the main menu to the footer. You still have access in any page (+ in the TX notifications) but in a less intrusive way. It will only show on dev env (local network), like the Faucet.

![image](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/7fcad03c-91cc-4cb6-a442-5bfab7cd438b)
![image](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/fcac12d6-8282-435c-baca-5fd770621a04)

---

Anyway, just tinkering. Let me know what you think!

carletex commented on packages/nextjs/components/Footer.tsx:
```diff
@@ -12,21 +14,32 @@ import { getTargetNetwork } from "~~/utils/scaffold-eth";
  */
 export const Footer = () => {
   const nativeCurrencyPrice = useGlobalState(state => state.nativeCurrencyPrice);
+  const isLocalNetwork = getTargetNetwork().id === hardhat.id;
```
I think we should probably move this util to `network.ts`, since we are using it in a bunch of places.

We could create and issue + another PR.

technophile-04 replied:
Yup agree ! Will create a PR for it 🙌

carletex commented on packages/nextjs/components/Footer.tsx:
```diff
@@ -12,21 +14,32 @@ import { getTargetNetwork } from "~~/utils/scaffold-eth";
-            {getTargetNetwork().id === hardhat.id && <Faucet />}
+            {isLocalNetwork && (
+              <>
+                <Faucet />
+                <Link href="/blockexplorer" passHref className="btn btn-primary btn-sm font-normal gap-0 normal-case">
+                  <MagnifyingGlassIcon className="h-4 w-4 mr-0.5" />
+                  <span>Block Explorer</span>
+                </Link>
```
A few variations here that we could try too:

- Only the Glass Icon
- Only :mag: Explorer
- Switch places with the Faucet...

rin-st replied:
- 🔍 ScaffoldEthScan  (or Se2-Scan) 😄

Pabl0cks replied:
I like it with the current icon + text 👌

technophile-04 replied:
Yeah even I like current sequence and icon + text 🙌

rin-st reviewed (approved):
I like the way it looks now 👍

Pabl0cks reviewed (approved):
Looking great to me 🙌

technophile-04 reviewed (approved):
Really looks clean !! and also now we don't have to manually remove `Block Explorer` it will be automatically removed if switched to other network 🙌

Tysm Carlos !! 

Nitpick : 

![Screenshot 2023-10-14 at 6 38 54 PM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/23ad9fda-b76d-4f18-9ea9-2f7e837e25a1)
 
 Maybe we should keep the same spacing b/w text icons for all buttons?

carletex commented:
> Maybe we should keep the same spacing b/w text icons for all buttons?

Changed to use the same classes (and removed some that weren't doing anything). Visually it doesn't look 100% aligned, because the 16x16 icons have some air around and the "bankNote" one is wider than the other two... also, the "1" gives the feeling of being more on the right than "F" or "B"

Anyway, this is how it looks:

![image](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/9a27b1a4-2d04-4d04-b842-69793355743d)

carletex commented:
Thanks all for the review!

Feel free to merge when checking the last commit.

Pabl0cks commented:
Looks great to me! But I'll let Shiv's with his hawk-eye have the last word 🧐

technophile-04 commented:
>  Visually it doesn't look 100% aligned, because the 16x16 icons have some air around and the "bankNote" one is wider than the other two... also, the "1" gives the feeling of being more on the right than "F" or "B"

Ohh yes, that makes sense! I think it looks fine 🙌


Also, SE-2 will be mostly viewed on mobile if it's in prod (i.e no `hardhat` network) so both Facuet and Blockexplorer shouldn't show up on mobile 🙈 lol sorry for my OCD kicking in 


Tysm Carlos 🙏 look great to me 🙌
